nexus_constructor/instrument_view.py

```diff
@@ -93,6 +94,24 @@ def add_component(self, name, geometry):
         self.component_meshes[name] = mesh
         self.component_entities[name] = entity
 
+    def delete_component(self, name):
+        """
+        Delete a component from the InstrumentView by removing the components and entities from the dictionaries.
+        :param name: The name of the component.
+        """
+        try:
+            del self.component_entities[name]
+            del self.component_meshes[name]
+            del self.component_translations[name]
+        except KeyError:
```

Comment:
stick a print statement here saying something like "unable to delete component" - just avoids other people having to guess where the code is failing in the future rather than failing silently